pkg/core/writer/signalfx/spans.go

```diff
@@ -21,9 +21,21 @@ func (sw *Writer) sendSpans(ctx context.Context, spans []*trace.Span) error {
 		// This sends synchonously
 		err := sw.client.AddSpans(context.Background(), spans)
 		if err != nil {
-			log.WithFields(log.Fields{
-				"error": err,
-			}).Error("Error shipping spans to SignalFx")
+			var meta log.Fields
+			if sw.conf.LogTraceSpansFailedToShip {
+				jsonEncodedSpans, _ := json.Marshal(spans)
```

Comment:
err should be checked or this can panic. Also if we failed to ship them it is possible json.Marshal could fail as well.

Reply:
I think `jsonEncodedSpans` would be nil if there were an err so payload below would just show nil.  A json marshal error is pretty hard to imagine though since we aren't using generic data structures in the spans, just a well-defined span format that we know serializes fine.